.github/scripts/query_observe.py

```diff
@@ -0,0 +1,299 @@
+import configparser
+import json
+import re
+import logging
+import requests
+import os, sys
+import datetime
+from datetime import timezone
+
+
+# Create a logger instance
+def setup_logger(log_level, log_format):
+    """
+    Setup logger with specified log level and format.
+
+    @param log_level: Logging level (default is DEBUG)
+    @param log_format: Logging format (default is '%(asctime)s - %(levelname)s - %(message)s')
+    """
+    logger.setLevel(log_level)
+    formatter = logging.Formatter(log_format)
+
+    # Create console handler and set level to debug
+    ch = logging.StreamHandler()
+    ch.setLevel(log_level)
+    ch.setFormatter(formatter)
+
+    # Add console handler to logger
+    logger.addHandler(ch)
+
+
+def get_bearer_token() -> str:
+    """Logins into account and gets bearer token
+    @return: bearer_token
+    """
+
+    customer_id = os.environ.get("OBSERVE_CUSTOMER")
+    domain = os.environ.get("OBSERVE_DOMAIN")
+    user_email = os.environ.get("OBSERVE_USER_EMAIL")
+    user_password = os.environ.get("OBSERVE_USER_PASSWORD")
+
+    url = f"https://{customer_id}.{domain}/v1/login"
+
+    message = '{"user_email":"$user_email$","user_password":"$user_password$", "tokenName":"terraform-azure-collection"}'
+    tokens_to_replace = {
+        "$user_email$": user_email,
+        "$user_password$": user_password,
+    }
+    for key, value in tokens_to_replace.items():
+        message = message.replace(key, value)
+
+    header = {
+        "Content-Type": "application/json",
+    }
+
+    response = json.loads(
+        requests.post(url, data=message, headers=header, timeout=10).text
+    )
+    bearer_token = response['access_key']
+    return bearer_token
+
+
+def send_query(bearer_token: str, query: str, params: dict = None, url_extension: str = '',
```

Comment:
same  for this code on reusability

Reply:
https://observe.atlassian.net/browse/OB-31450
For future